src/dc_receive_imf.rs

```diff
@@ -802,11 +802,21 @@ async fn add_parts(
     let mut parts = std::mem::replace(&mut mime_parser.parts, Vec::new());
     let server_folder = server_folder.as_ref().to_string();
     let is_system_message = mime_parser.is_system_message;
-    let mime_headers = if save_mime_headers {
+
+    // if needed, we save the mime-message for the first part containing text
+    // (ui typically displays a button to load whole message)
+    //
+    // (currently, we skip that part for encrypted messages
+    // as there is probably no huge intersection between html-messages and encrypted messages,
+    // however, that should be doable we need the decrypted mime-structure in this case, however)
+    let mut save_mime_modified = mime_parser.is_mime_modified && !mime_parser.was_encrypted();
```

Comment:
Maybe invert the logic, and initialize it to `let mut mime_modified_stored = mime_parser.encrypted()`. Then set it to `true` once it has been set once.

Reply:
i'll refine the documentation. inverting the logic would require to use/check two booleans (mime_modified_saved and shall_save_modified or so), so that probably would not add much overview.